### DISCUSSION TITLE
courses/beginners-web-handbook/carousel-project-1/12-javascript-10

### DISCUSSION THREAD

**Discussion by giscus** [crwdu/lms-contents-js101#40] [Q&A]
# courses/beginners-web-handbook/carousel-project-1/12-javascript-10

바닐라코딩

https://www.vcdr.club/courses/beginners-web-handbook/carousel-project-1/12-javascript-10

**Comment by jin-ttao** [comment]
질문할 부분) 마지막 const lists = document.querySelectorAll("li"); 관련

질문) querySelectorAll 로 선택자가 찾은 요소들이 배열로 변수에 들어가는지, NodeList로 들어가는지 궁금해요. 바로 위 "lists 변수에 배열 형태로 담긴다"는 부분을 이해하지 못해서 찾아보다가, querySelectorAll의 결과물로 (배열과 유사하지만 다른 종류의) NodeList 형태가 된다는 내용이 있어 질문 드려요. (+초반 학습이라 이런 부분은 대략적으로만 이해하고 넘어가는 방향이 좋을까요??)

**Reply by Ken123777** [reply]
입문하시는 분들께서 이해하기에 무리가 있지 않도록 배열이라고 표현해놓았으나, 사실 정확하게는 조사하신 것처럼 NodeList가 맞습니다. 다만, 배열과의 차이점을 조사 해보시되 현재로서는 이해가 되지 않는 부분이 있더라도 넘어가셔도 좋을것 같습니다. 🙂
